tools/mason/MasonPrereqs.chpl

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020-2025 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FileSystem as FS;
+use FileSystem only cwd,chdir; // these are tertiary methods on locale
+import Path;
+import List.list;
+
+import MasonLogger;
+import MasonUtils;
+
+private config param makeTargetChplFlags = "printchplflags";
+
+private var log = new MasonLogger.logger("mason prereqs");
+
+record pushdMgr: contextManager {
+  var path: string;
+  var initDir = here.cwd();
+
+  proc ref enterContext() { here.chdir(path); }
+  proc ref exitContext(in err: owned Error?) throws {
+    if err then throw err;
+    here.chdir(initDir);
+  }
+}
+
+private inline proc pushd(path: string) {
+  return new pushdMgr(path);
+}
+
+proc install(path: string) {
+  log.infof("Installing prerequisite %s\n", path);
+  manage pushd(path) {
+    MasonUtils.runCommand("make"); // TODO check for errors
```

Comment:
I think it would be really nice to allow more than just `make`. Would you open an issue for that?

Reply:
Ah, I see. Yeah, that sounds cool. I'll fire up an issue with some of my thoughts.

Reply:
https://github.com/chapel-lang/chapel/issues/28174 is the issue for this.